### PR TITLE
Add repo settings.

### PR DESCRIPTION
**Pull request by jcape**
- Add settings.yml
- Protect access to settings.yml and CODEOWNERS

### Motivation

This PR applies settings.yml to this repository, in the hope that we're able to also work it into the templates.

### CODE REVIEW

**Comment by jcape** [comment]
Current dependencies on/for this PR:
* main
  * **PR #15** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/cookiecutters/15" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/cookiecutters/15?utm_source=stack-comment).

File: .github/settings.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  - name: size/XS
+    color: '#00ed01'
+    description: Extra-Small PRs
+  - name: size/S
+    color: '#3af901'
+    description: Small PRs
+  - name: size/M
+    color: '#cefb02'
+    description Medium-sized PRs
```
Comment (nick-mobilecoin) [inline .github/settings.yml:38]: :wrench: 
```suggestion
    description: Medium-sized PRs
```

File: .github/settings.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      required_pull_request_reviews:
+        require_code_owner_reviews: true
+        required_approving_review_count: 1
+        require_last_push_approval: true
+      restrictions:
+        apps: []
+        users: []
+        teams: []
+      required_linear_history: true
+      required_conversation_resolution: true
```
Comment (nick-mobilecoin) [inline .github/settings.yml:78]: :question: Did we want required signatures?
```suggestion
      required_conversation_resolution: true
      required_signatures: true
```

**Review by nick-mobilecoin — CHANGES_REQUESTED** [review]
Just need to fix the one label

**Review by nick-mobilecoin — APPROVED** [review]

**Pull request merged**